src/api/offline/ledgerhash.js

```diff
@@ -28,25 +28,47 @@ function hashLedgerHeader(ledgerHeader) {
   return common.core.Ledger.calculateLedgerHash(header);
 }
 
-function computeLedgerHash(ledgerHeader: Object, transactions: Array<Object>
-): string {
-  if (transactions) {
-    const txs = _.map(transactions, tx => {
-      const mergeTx = _.assign({}, _.omit(tx, 'tx'), tx.tx || {});
-      const renameMeta = _.assign({}, _.omit(mergeTx, 'meta'),
-        tx.meta ? {metaData: tx.meta} : {});
-      return renameMeta;
-    });
-    const ledger = common.core.Ledger.from_json({transactions: txs});
-    const transactionHash = ledger.calc_tx_hash().to_hex();
-    if (ledgerHeader.transaction_hash !== undefined
-        && ledgerHeader.transaction_hash !== transactionHash) {
-      throw new common.errors.ValidationError('transaction_hash in header'
-        + ' does not match computed hash of transactions');
-    }
-    return hashLedgerHeader(_.assign({}, ledgerHeader, {transactionHash}));
+function computeTransactionHash(ledger) {
+  if (ledger.rawTransactions === undefined) {
+    return ledger.transactionHash;
   }
-  return hashLedgerHeader(ledgerHeader);
+  const transactions = JSON.parse(ledger.rawTransactions);
+  const txs = _.map(transactions, tx => {
+    const mergeTx = _.assign({}, _.omit(tx, 'tx'), tx.tx || {});
+    const renameMeta = _.assign({}, _.omit(mergeTx, 'meta'),
+      tx.meta ? {metaData: tx.meta} : {});
+    return renameMeta;
+  });
+  const ledgerObject = common.core.Ledger.from_json({transactions: txs});
+  const transactionHash = ledgerObject.calc_tx_hash().to_hex();
+  if (ledger.transactionHash !== undefined
+      && ledger.transactionHash !== transactionHash) {
+    throw new common.errors.ValidationError('transactionHash in header'
+      + ' does not match computed hash of transactions');
+  }
+  return transactionHash;
+}
+
+function computeStateHash(ledger) {
+  if (ledger.rawState === undefined) {
+    return ledger.stateHash;
+  }
+  const state = JSON.parse(ledger.rawState);
+  const ledgerObject = common.core.Ledger.from_json({accountState: state});
+  const stateHash = ledgerObject.calc_account_hash().to_hex();
+  if (ledger.stateHash !== undefined && ledger.stateHash !== stateHash) {
+    throw new common.errors.ValidationError('stateHash in header'
+      + ' does not match computed hash of state');
+  }
+  return stateHash;
+}
+
+function computeLedgerHash(ledger: Object): string {
```

Comment:
I'm just suggesting something simple for the moment, like keeping the `RPC
handlers` grouped with a comment. You don't have to go so far as creating a
separate map, to extend RippleAPI.prototype from with the fancy _.assign
magic etc.

i.e. Just draw some lines around the shapes that are emerging so people
know where you're going. Broadly you have 3 classes of routines here,
future rpc handlers, rpc client methods, and utils for working with the new
api formats. You've said there's Wallet utils, and at some point XRPValue &
IOUAmount will be exposed, but maybe from another repo.

I guess it might not hurt to pull out your crystal ball a bit and try and
work out what sort of features you plan to expose?